massgov/openmass
# DP-19675 Add Linked content to text areas.

**Description:**
This is the initial framework for adding WYSIWG (textarea) support to the "Pages Linking Here" functionality.

This currently works with all of the WYSIWYG fields that I pulled out of every content type and related paragraph.  It is configured by adding those fields to the node.type.{content_type}.yml file. in the mass_content_api linking_pages section.

**Jira:**
https://jira.mass.gov/browse/DP-19675


**To Test:**
- [ ] View this tab: /news/baker-polito-administration-announces-covid-19-vaccine-advisory-group/linking-page
- [ ] Browse here and click save: http://mass.local/info-details/covid-19-vaccine-frequently-asked-questions/edit
- [ ] Trigger a cron run to insure the queues get run.
- [ ] Repeat step 1.

_If you completely run the mass_content_api_descendant_queue first, you will not see a difference. when running the steps above, since the queue run already queued up the FAQ node for re-indexing of descendant content._ 

- [ ] Find a node that you'd like to link to.
- [ ] Copy and paste the local relative URL (e.g. "/info-details/covid-19-vaccinations-for-health-care-workers")
- [ ] Browse to another unrelated page (e.g. "/how-to/create-your-masstaxconnect-individual-account")
- [ ] Edit the node, and in a WYSIWYG Field, link some text to the first page.
- [ ] Trigger a cron run to insure the queues get run.
- [ ] View the linked pages from the node you edited, and see the first page listed there.

**Screenshots/GIFs:**

Other content is also showing the basic functionality as well.
![image](https://user-images.githubusercontent.com/376224/109273380-5d2b8180-77cf-11eb-8f59-1c4c85716fc3.png)


---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)
